crates/crash-handler/src/lib.rs

```diff
@@ -44,6 +45,11 @@ fn handle_panic(panic_info: &PanicInfo<'_>) {
     // Wait till the logs have been flushed
     aptos_logger::flush();
 
+    // Do not kill the process if the panics happened at move-bytecode-verifier.
+    if state::get_state() == VMState::VERIFIER {
```

Comment:
Nice! 😄 Question: I'm not sure how the state is managed on the `move-core-types` side, but are there any concerns around race conditions here? e.g., what happens if another part of the code panics while we're in the verification stage? Will this catch the error?

Reply:
A brief look at https://github.com/move-language/move/commit/e7e42d9ec1a689cc29c9142fcedd092413e3b59e suggests we should be okay? e.g., we're using thread local storage to set this global variable. And, I assume that that the verifier is not going to context switch away during verification (e.g., it doesn't use any async/await), so nothing else should panic during the verification stage?

Reply:
Actually this seems a legit concern here? If another thread panics during the verification we should read out the `VERIFIER` state and thus skip the panic?

Reply:
That was a great call! I have a reproducing example here: https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=3cadcce23944a0b7d326779c921d0022. This demonstrates that each thread is usinge its own global variable so no race condition could happen.

Reply:
The beauty of thread locals is that there are no race conditions. However, this reminds of another related concern. Was is if the crash happens in a synchronously called function `f` from an external crate via the verifier? We had long discussions about this also with OtterSec. As long as the called code is [UnwindSave](https://doc.rust-lang.org/std/panic/trait.UnwindSafe.html) we should be good. At least we know that the bytecode verifier only calls core Rust framework code which expect to be UnwindSafe. So what we are doing should only be done for certain primitive kind of code.